Modules/rat_board.py

```diff
@@ -199,10 +199,43 @@ def find_by_uuid(self, guid: UUID) -> Optional[Rescue]:
 
         """
         for rescue in self.rescues.values():
-            if rescue.case_id == guid:
+            if rescue.uuid == guid:
                 return rescue
         return None
 
+    def search(self, case: str) -> Optional[Rescue]:
+        """
+        Helper for searching for a case using implemented search functions.
+
+        Args:
+            case str: The case string to search on
+
+        Returns:
+            Rescue: found rescue
+            None:   No rescue found
+
+        Raises:
+            ValueError: argument was not an expected type
+        """
+        # ensure a proper type was passed
+        if not isinstance(case, str):
+            raise ValueError(f"Expected str, got {type(case)} instead")
+        # strings could be any of the valid types, check for the best one.
+        if case.isdigit():
+            return self.find_by_index(int(case))
+        # UUIDs always start with @ and are 37 characters, so try to parse it
+        elif case[0] == "@" and len(case) == 37:
+            try:
+                guid_str = case[1:]
+                guid = UUID(guid_str)
+            except (AttributeError, ValueError):
+                # not a valid UUID
+                raise ValueError("Specified uuid is not valid.")
```

Comment:
catch/re-raise only muddles the stacktrace.  which makes debugging more difficult. i suggest not catching it at all